### PR TITLE
Fixes #17403: refactoring io usage and configuration

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17403

### CODE REVIEW

**Comment by gpoblon** [comment]
still requires some work to properly handle other Generators

File: rudder-lang/src/io.rs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    } else if config.action == Some(Action::Translate) && fmt == "rl" {
+        Ok(fmt.to_owned())
+    } else {
+        Err(Error::User(
+            format!("Could not determine format: {} is not a valid format", fmt),
+        ))
+    }
+}
+
+fn get_opt_action_mode(action: Action, config: &Config) -> Result<ActionConfig> {
```
Comment (amousset) [inline rudder-lang/src/io.rs:173]: why returning a result here?

Comment (gpoblon) [reply rudder-lang/src/io.rs:173]: No point doing it. I will fix it

**Comment by amousset** [comment]
Why removing execution bit for  rudder-lang/tools/tester.sh?

**Comment by gpoblon** [comment]
I will roll it back to 755

**Comment by gpoblon** [comment]
fixed

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**